src/main/java/ch/njol/skript/effects/EffKill.java

```diff
@@ -26,40 +26,46 @@
 		"Note: This effect does not set the entity's health to 0 (which causes issues), but damages the entity by 100 times its maximum health."})
 @Examples({"kill the player",
 		"kill all creepers in the player's world",
-		"kill all endermen, witches and bats"})
-@Since("1.0")
+		"kill all endermen, witches and bats",
+		"kill the player ignoring totem of undying",
+		"kill target entity ignoring totem of undying",
+})
+@Since("1.0, INSERT VERSION (ignoring totem of undying)")
 public class EffKill extends Effect {
 
 	static {
-		Skript.registerEffect(EffKill.class, "kill %entities%");
+		Skript.registerEffect(EffKill.class, "kill %entities%", "kill %entities% ignoring [a|any] totem[s] of undying");
 	}
 	
-	// Absolutely make sure it dies
-	public static final int DAMAGE_AMOUNT = Integer.MAX_VALUE;
-	
 	@SuppressWarnings("null")
 	private Expression<Entity> entities;
+	private boolean ignoreTotem;
 	
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
 		entities = (Expression<Entity>) exprs[0];
+		ignoreTotem = matchedPattern == 1;
 		return true;
 	}
 
 	@Override
-	protected void execute(final Event e) {
-		for (Entity entity : entities.getArray(e)) {
+	protected void execute(Event event) {
+		for (Entity entity : entities.getArray(event)) {
 
 			if (entity instanceof EnderDragonPart) {
 				entity = ((EnderDragonPart) entity).getParent();
 			}
 
+			if (ignoreTotem) {
+				HealthUtils.setHealth((Damageable) entity, 0);
+				return;
+			}
+
 			if (entity instanceof Damageable) {
-				final boolean creative = entity instanceof Player && ((Player) entity).getGameMode() == GameMode.CREATIVE;
+				boolean creative = entity instanceof Player player && player.getGameMode() == GameMode.CREATIVE;
 				if (creative) // Set player to survival before applying damage
 					((Player) entity).setGameMode(GameMode.SURVIVAL);
```

Comment:
```suggestion
					player.setGameMode(GameMode.SURVIVAL);
```

Reply:
Cannot be changed, it does not accept `player`, as seen below
![image](https://github.com/user-attachments/assets/a3cc9f4f-d4ea-4af1-9007-76cb0a554624)